### PR TITLE
#1 - Pre-release tidy

### PR DESCRIPTION
**Pull request by devinivy**
- Upgrades deps, removes unused deps
 - Consistency in readme, package.json, added license, etc.
 - Makes `skippedCols` output structured
 - No more use of (module-)global variable for `skippedCols`
 - Use new handlebars instance rather than the global one
 - Fix some checks in tests
 - Basic cleanup

Still on my mind,
 1. It seems like skipped columns are not omitted from the model side of the diff, so skipped columns can still appear in migrations.  This is a bug I think we should fix before the release.
 2. Would feel great to not rely on the semi-colon regex hack.
 3. The dynamic tests are acting a little weird—there's no way to mark one as `only` and lab doesn't know about them at runtime, so you can't reference them by id (e.g. `lab -i 24`).  I think this is fixable.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ internals.cleanup = (session, done) => {
                 Fs.unlinkSync(filePath);
             });
 
-        session.knex.destroy().asCallback(done);
+        session.knex.destroy().then(() => done(), done);
```
Comment (wswoodruff) [inline test/index.js]: What lead to this change? I thought asCallback would pass the error to done and we'd be able to get it from there. If destroy() fails what happens now?

**Review by wswoodruff — APPROVED** [review]
Nice tidying, thank you!!

**Comment by wswoodruff** [comment]
Re: Description notes:
1. I think a hard error is thrown if there's an unsupported Joi prop, are you thinking it'd be better to list it in skippedCols output instead?
2. Open to ideas -- using straight logic to figure out when the last part of the promise chain was being written was a big mess.
3. Also open to ideas

File: test/index.js (hunk 1 above)
Comment (devinivy) [reply test/index.js]: You're totally right—upgrading knex broke this https://github.com/tgriesser/knex/pull/2589

**Pull request merged**